external-crates/move/crates/move-analyzer/src/symbols.rs

```diff
@@ -2139,46 +2231,67 @@ impl<'a> TypingSymbolicator<'a> {
                 ),
             );
 
-            self.struct_symbols(s);
+            self.struct_symbols(s, name, mod_ident_str);
         }
         self.use_funs_symbols(&mod_def.use_funs);
     }
 
     /// Get symbols for struct definition
-    fn struct_symbols(&mut self, struct_def: &StructDefinition) {
+    fn struct_symbols(
+        &mut self,
+        struct_def: &StructDefinition,
+        struct_name: &Symbol,
+        mod_ident_str: &String,
+    ) {
         // create scope designated to contain type parameters (if any)
         let mut tp_scope = BTreeMap::new();
         for stp in &struct_def.type_parameters {
             self.add_type_param(&stp.param, &mut tp_scope);
         }
+
+        let positional = self
+            .mod_outer_defs
+            .get(mod_ident_str)
+            .map_or(false, |mod_defs| {
+                mod_defs
+                    .structs
+                    .get(struct_name)
+                    .map_or(false, |sdef| sdef.positional)
+            });
+
         self.type_params = tp_scope;
         if let StructFields::Defined(fields) = &struct_def.fields {
             for (fpos, fname, (_, t)) in fields {
                 self.add_type_id_use_def(t);
-                // enter self-definition for field name (unwrap safe - done when inserting def)
-                let start = get_start_loc(&fpos, self.files, self.file_id_mapping).unwrap();
-                let field_info = DefInfo::Type(t.clone());
-                let ident_type_def_loc = def_info_to_type_def_loc(self.mod_outer_defs, &field_info);
-                let doc_string = extract_doc_string(
-                    self.file_id_mapping,
-                    self.file_id_to_lines,
-                    &start,
-                    &fpos.file_hash(),
-                );
-                self.use_defs.insert(
-                    start.line,
-                    UseDef::new(
-                        self.references,
-                        self.alias_lengths,
-                        fpos.file_hash(),
-                        start,
-                        fpos.file_hash(),
-                        start,
-                        fname,
-                        ident_type_def_loc,
-                        doc_string,
-                    ),
-                );
+                if !positional {
+                    // enter self-definition for field name (unwrap safe - done when inserting def),
+                    // but only if the fields are named (positional fields have "fake" locations and
+                    // could make the displayed results confusing)
```

Comment:
Do you think it's worth mentioning that positional fields are related to Move 2024 here? Just think it might be tricky to understand why positional fields would have fake locations when looking at this comment in isolation. But it's minor